src/main/scala/io/github/pauljamescleary/petstore/domain/users/UserValidationInterpreter.scala

```diff
@@ -4,22 +4,25 @@ package users
 import cats._
 import cats.data.EitherT
 import cats.implicits._
+import Function._
 
 class UserValidationInterpreter[F[_]: Monad](userRepo: UserRepositoryAlgebra[F]) extends UserValidationAlgebra[F] {
-  def doesNotExist(user: User) = EitherT {
-    userRepo.findByUserName(user.userName).map {
-      case None => Right(())
-      case Some(_) => Left(UserAlreadyExistsError(user))
+  def doesNotExist(user: User): EitherT[F, UserAlreadyExistsError, Unit] =
```

Comment:
looks like this doesn't change anything, but this method and others in this file should be simplified, using `.as` and `.toLeft` from `OptionT`.

Reply:
I favour simplicity and easy to understand code over shortness.
And I think the code from below is harder to understand than the one we already have. I am not a big fan of this kind of not primitive operations.
```scala
def doesNotExist(user: User): EitherT[F, UserAlreadyExistsError, Unit] =
    userRepo
      .findByUserName(user.userName)
      .map(UserAlreadyExistsError)
      .toLeft(())
```

Reply:
I will change it for the latter though.